### ISSUE TITLE
Test simpler and more correct actions/checkout

### ISSUE THREAD

**Issue by kimgr** [include-what-you-use/include-what-you-use#1506]
[This issue](https://github.com/actions/checkout/issues/15) mentions that actions/checkout by default checks out the so-called "merge commit" for PRs -- I think that means latest master + PR patch, which would be much nicer for contributors rather than having to manually rebase for correct CI.

Presumably it also does the right thing for non-pull-request events, so we can get rid of this special handling:
https://github.com/include-what-you-use/include-what-you-use/blob/693d98e019cfbaa5c03660fcefb984eae7c58dbe/.github/workflows/ci.yml#L45-L58

Experiment and see if it can be made to work.

**Comment by firewave** [comment]
Copying my comment from https://github.com/include-what-you-use/include-what-you-use/pull/1502#issuecomment-2034241733:

I see no need for that `IWYU_CHECKOUT_REF` stuff. Just do `uses: actions/checkout@v4` without any parameters.

**Comment by kimgr** [comment]
Thanks! I still stand by my commit message: https://github.com/include-what-you-use/include-what-you-use/commit/44fa8c7bd042907dd7bb9a42dbefdf1ad346c313 -- I have no clue what `[...] the last merge commit of the pull request merge branch` means, but I'm willing to risk it :)

**Comment by kimgr** [comment]
Ah, now I think I know why I went to the trouble -- we try to get the current sha to bake it into the IWYU build, and IIRC that step failed. I'll try again and see what falls out.

**Comment by kimgr** [comment]
@firewave If you have a chance, please verify that I've understood the world correctly: https://github.com/include-what-you-use/include-what-you-use/pull/1508

**Comment by firewave** [comment]
Err - why handle merge commits, if you aren't using them? Or am I missing something?

**Comment by kimgr** [comment]
My mega commit message just tries to explain what GitHub does behind the scenes, not anything we're doing.

GitHub creates a new "ref" for pull requests, which is kind-of like a branch, but doesn't show up in branch listings. E.g. for 1508 this ref is available at: https://github.com/include-what-you-use/include-what-you-use/tree/pull/1508/merge.

You'll note that the top commit, 311b0c97d55389f3d34b8800081be86e2fed7de7, says:
```
Merge 5efa3bd into f54135c

kimgr committed 20 hours ago
```

I've done no such thing, it's just an ephemeral merge that GitHub does to put the PR back on top of master. Content-wise it's identical to a fully rebased PR.

I wasn't fully aware of this behavior before, but I noticed that none of my commit SHAs lined up, which is why I used the `IWYU_CHECKOUT_REF` trick to leave the PR as the author provided it.
